src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1929,12 +1929,19 @@ private CSDiagnosticInfo NotFound(SyntaxNode where, string simpleName, int arity
                         fullName = qualifierOpt.ToDisplayString(SymbolDisplayFormat.QualifiedNameOnlyFormat) + "." + fullName;
                     }
 
-                    forwardedToAssembly = GetForwardedToAssembly(fullName, arity, out encounteredForwardingCycle);
+                    forwardedToAssembly = GetForwardedToAssembly(fullName, arity, out encounteredForwardingError);
 
-                    if (encounteredForwardingCycle)
+                    if (encounteredForwardingError != null)
                     {
-                        Debug.Assert((object)forwardedToAssembly != null, "How did we find a cycle if there was no forwarding?");
-                        diagnostics.Add(ErrorCode.ERR_CycleInTypeForwarder, location, fullName, forwardedToAssembly.Name);
+                        if (encounteredForwardingError.Code == (int)ErrorCode.ERR_CycleInTypeForwarder)
+                        {
+                            Debug.Assert((object)forwardedToAssembly != null, "How did we find a cycle if there was no forwarding?");
+                            diagnostics.Add(ErrorCode.ERR_CycleInTypeForwarder, location, fullName, forwardedToAssembly.Name);
```

Comment:
Why are we creating new diagnostic infos instead of using the ones we already have. If the ones we have a re missing something, perhaps we shouldn't be creating them in GetForwardedToAssembly, but should simply return the error number back.